File: app/src/main/java/com/worldventures/dreamtrips/modules/membership/model/converter/InviteTemplateFromInvitationPreviewConverter.java
package com.worldventures.dreamtrips.modules.membership.model.converter;


import com.worldventures.dreamtrips.api.invitation.model.InvitationPreview;
import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.modules.membership.model.InviteTemplate;

import io.techery.mappery.MapperyContext;

public class InviteTemplateFromInvitationPreviewConverter implements Converter<InvitationPreview, InviteTemplate> {
   @Override
   public Class<InvitationPreview> sourceClass() {
      return InvitationPreview.class;
   }

   @Override
   public Class<InviteTemplate> targetClass() {
      return InviteTemplate.class;
   }

   @Override
   public InviteTemplate convert(MapperyContext mapperyContext, InvitationPreview invitationTemplate) {
      return new InviteTemplate(invitationTemplate.id(), invitationTemplate.title(),
            invitationTemplate.coverImage().url(), invitationTemplate.video(),
            invitationTemplate.link(), invitationTemplate.locale(), invitationTemplate.content());
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/domain/converter/InviteTemplateFromInvitationPreviewConverter.kt
package com.worldventures.dreamtrips.social.domain.converter


import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.invitation.model.InvitationPreview
import com.worldventures.dreamtrips.social.domain.entity.InviteTemplate

import io.techery.mappery.MapperyContext

class InviteTemplateFromInvitationPreviewConverter : Converter<InvitationPreview, InviteTemplate> {

   override fun sourceClass() = InvitationPreview::class.java

   override fun targetClass() = InviteTemplate::class.java

   override fun convert(context: MapperyContext, source: InvitationPreview) =
         InviteTemplate(source.id(), source.title(), source.coverImage().url(), source.content(), null, source.link())
}
